x-pack/plugins/apm/public/components/app/service_details/service_icons/container_details.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiDescriptionList } from '@elastic/eui';
+import { EuiDescriptionListProps } from '@elastic/eui/src/components/description_list/description_list';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { APIReturnType } from '../../../../services/rest/createCallApmApi';
+
+type ServiceDetailsReturnType = APIReturnType<'GET /api/apm/services/{serviceName}/metadata/details'>;
+
+interface Props {
+  container: ServiceDetailsReturnType['container'];
+}
+
+export function ContainerDetails({ container }: Props) {
+  if (!container) {
+    return null;
+  }
+
+  const listItems: EuiDescriptionListProps['listItems'] = [];
+  if (container.os) {
+    listItems.push({
+      title: i18n.translate('xpack.apm.serviceNameHeader.container.os', {
+        defaultMessage: 'OS',
+      }),
+      description: container.os,
+    });
+  }
+
+  if (container.isContainerized !== undefined) {
+    listItems.push({
+      title: i18n.translate(
+        'xpack.apm.serviceNameHeader.container.containerized',
+        { defaultMessage: 'Containerized' }
+      ),
+      description: container.isContainerized
+        ? i18n.translate('xpack.apm.serviceNameHeader.container.yes', {
+            defaultMessage: 'Yes',
+          })
+        : i18n.translate('xpack.apm.serviceNameHeader.container.no', {
+            defaultMessage: 'No',
+          }),
+    });
+  }
+
+  if (container.totalNumberInstances) {
+    listItems.push({
+      title: i18n.translate(
+        'xpack.apm.serviceNameHeader.container.totalNumberInstances',
+        { defaultMessage: 'Total number of instances' }
+      ),
+      description: Math.round(container.totalNumberInstances),
```

Comment:
nit: Let's use the existing formatters when possible
```suggestion
      description: asInteger(container.totalNumberInstances),
```